awx/main/models/credential/__init__.py

```diff
@@ -1187,6 +1186,37 @@ def create(self):
     }
 )
 
+ManagedCredentialType(
+    namespace='registry',
+    kind='registry',
+    name=ugettext_noop('Container Registry'),
+    inputs={
+        'fields': [{
+            'id': 'host',
+            'label': ugettext_noop('Authentication URL'),
+            'type': 'string',
+            'help_text': ugettext_noop('Authentication endpoint for the container registry.'),
+        }, {
+            'id': 'username',
+            'label': ugettext_noop('Username'),
+            'type': 'string',
+        }, {
+            'id': 'password',
+            'label': ugettext_noop('Password'),
+            'type': 'string',
+            'secret': True,
+        }, {
+            'id': 'token',
```

Comment:
So this isn't hooked up yet, I'd like to get some definition around that. What I've seen before is

```
podman login -u='user' -p='token'
```

That's used for pushing. So obviously you need your token to push to quay.io. However, I have not experimented with pulling from quay private repos or anything like that. I'm also unclear how this will work with Automation Hub authentication, to use their image registry, and the details of that authentication are certain to be different. But this is just 1 credential type, so how would we deal with that? And does it even work?

Reply:
The three registries I looked into all take either username+password or a token.  If someone has specific knowledge of what changes are needed here to do the right thing, I would love to hear it.